File: packages/superwall/android/src/main/java/io/capawesome/capacitorjs/plugins/superwall/classes/options/HandleDeepLinkOptions.java
package io.capawesome.capacitorjs.plugins.superwall.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.superwall.classes.CustomExceptions;

public class HandleDeepLinkOptions {

    @NonNull
    private final String url;

    public HandleDeepLinkOptions(@NonNull PluginCall call) throws Exception {
        this.url = getUrlFromCall(call);
    }

    @NonNull
    public String getUrl() {
        return url;
    }

    @NonNull
    private static String getUrlFromCall(@NonNull PluginCall call) throws Exception {
        String url = call.getString("url");
        if (url == null) {
            throw CustomExceptions.URL_MISSING;
        }
        return url;
    }
}

File: packages/superwall/ios/Plugin/Classes/Options/HandleDeepLinkOptions.swift
import Foundation
import Capacitor

@objc public class HandleDeepLinkOptions: NSObject {
    let url: String

    init(_ call: CAPPluginCall) throws {
        self.url = try HandleDeepLinkOptions.getUrlFromCall(call)
    }

    private static func getUrlFromCall(_ call: CAPPluginCall) throws -> String {
        guard let url = call.getString("url") else {
            throw CustomError.urlMissing
        }
        return url
    }
}
